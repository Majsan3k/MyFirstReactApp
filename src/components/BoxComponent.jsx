import React, {Component} from 'react';
import PropTypes from 'prop-types';

export default class BoxComponent extends Component{

    state = {
        clicked : false
    };

    static propTypes = {
      value: PropTypes.number
    };


    handleCollision = (otherComponentValue) => {
        this.setState(prevState => {
            return {value: prevState.value + otherComponentValue}

        });
    };

    resetBox = () => {
        this.setState({value: ''})
    };

    render(){
        return (
            <div className="box-not-clicked">
                <p>{this.props.value}</p>
            </div>
        )
    }
}