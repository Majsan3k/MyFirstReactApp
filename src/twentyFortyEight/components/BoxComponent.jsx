import React, {Component} from 'react';
import PropTypes from 'prop-types';

export default class BoxComponent extends Component {

    state = {
        value: this.props.value,
        id: this.props.id
    };

    static propTypes = {
        id: PropTypes.number.isRequired,
        value: PropTypes.any.isRequired,
        lastMove: PropTypes.string.isRequired
    };

    componentDidUpdate(prevProps) {
        if (prevProps.value !== this.props.value) {
            this.setState({value: this.props.value, clicked: false});
        }
    };

    render() {
        let classNames = require('classnames');
        let boxClass = classNames('box', {'no-value': this.state.value === ''});

        return (
            <div className={boxClass}>
                {this.state.value > 0 && <p>{this.state.value}</p>}
            </div>
        )
    }
}