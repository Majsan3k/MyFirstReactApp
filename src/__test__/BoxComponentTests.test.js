import React from 'react';
import BoxComponent from '../twentyFortyEight/components/BoxComponent';
import {render, fireEvent, cleanup} from 'react-testing-library';
import {mount} from 'enzyme'
import 'jest-dom/extend-expect'
import renderer from 'react-test-renderer';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('BoxComponent', () => {

    let box;
    let handleBoxClicked;

    beforeEach(() => {
        handleBoxClicked = jest.fn();
        const {container} = render(<BoxComponent
            value={'2'}
        />);
        box = container.querySelector('div');
    });

    afterEach(cleanup);

    it('should render correctly', () => {
        const {container} = render(<BoxComponent
            id={1}
            value={'2'}
            handleCollision={() => jest.fn()}
            boxClicked={''}
        />);
        expect(container).toMatchSnapshot();
    });
});

/* OLD TESTS - Used when a box still was clickable */

    /*
    it('should change class while clicked', () => {
        expect(box).toHaveClass('box not-clicked');
        fireEvent.click(box);
        expect(box).toHaveClass('box clicked');
    });

    it('should call handleBoxClicked while clicked', () => {
        fireEvent.click(box);
        expect(handleBoxClicked.mock.calls.length).toBe(1);
    });

    it('should call handleBoxClicked while clicked, test with enzyme-mount', () => {

        const handleBoxClickedMount = jest.fn();
        const wrapper = mount(<BoxComponent
            value={'2'}
            resetBoxClicked={jest.fn()}
            handleBoxClicked={handleBoxClickedMount}
            boxClicked={''}
        />);

        const spy = jest.spyOn(wrapper.instance(), 'handleClick');
        wrapper.instance().forceUpdate();
        wrapper.update();

        wrapper.find('div').simulate('click');
        expect(handleBoxClickedMount.mock.calls.length).toBe(1);

        expect(spy).toHaveBeenCalled();
    }); */
