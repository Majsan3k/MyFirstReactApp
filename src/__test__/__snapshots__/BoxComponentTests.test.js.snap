// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxComponent should render correctly 1`] = `
<div>
  <div
    class="box"
  >
    <p>
      2
    </p>
  </div>
</div>
`;
